staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -138,6 +144,56 @@ func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, re
 	return wc
 }
 
+// getStartWatchResourceVersion returns a ResourceVersion
+// the watch will be started from.
+// Depending on the input parameters the semantics of the returned ResourceVersion are:
+//   - start at Exact (return resourceVersion)
+//   - start at Most Recent (return an RV from etcd)
+func (w *watcher) getStartWatchResourceVersion(ctx context.Context, resourceVersion int64, opts storage.ListOptions) (int64, error) {
+	if resourceVersion > 0 {
+		return resourceVersion, nil
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
```

Comment:
Starting from this point we know that `resourceVersion=0`.

To make that more explicit, can we switch to `return 0, nil` below to make it more explicit

Reply:
sure, we could do that.